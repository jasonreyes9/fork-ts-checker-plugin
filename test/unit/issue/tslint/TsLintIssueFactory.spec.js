const {
  createIssueFromTsLintRuleFailure,
  createIssuesFromTsLintRuleFailures
} = require('../../../../lib/issue/tslint/TsLintIssueFactory');

describe('[UNIT] issue/tslint/TsLintIssueFactory', () => {
  const TS_LINT_RULE_FAILURE_WARNING = {
    getRuleName: () => 'no-unused-vars',
    getRuleSeverity: () => 'warning',
    getFailure: () => `'x' is assigned a value but never used.`,
    getFileName: () => 'src/test.ts',
    getStartPosition: () => ({
      getLineAndCharacter: () => ({
        line: 2,
        character: 2
      })
    })
  };
  const TS_LINT_RULE_FAILURE_ERROR = {
    getRuleName: () => 'no-unused-vars',
    getRuleSeverity: () => 'error',
    getFailure: () => `'y' is assigned a value but never used.`,
    getFileName: () => 'src/index.ts',
    getStartPosition: () => ({
      getLineAndCharacter: () => ({
        line: 10,
        character: 14
      })
    })
  };

  it.each([[TS_LINT_RULE_FAILURE_WARNING], [TS_LINT_RULE_FAILURE_ERROR]])(
    'creates Issue from TsLint RuleFailure: %p',
    ruleFailure => {
      const issue = createIssueFromTsLintRuleFailure(ruleFailure);

      expect(issue).toMatchSnapshot();
    }
  );

  it.each([[[TS_LINT_RULE_FAILURE_WARNING, TS_LINT_RULE_FAILURE_ERROR]]])(
    'creates Issues from TsLint RuleFailures: %p',
    ruleFailures => {
      const issues = createIssuesFromTsLintRuleFailures(ruleFailures);

      expect(issues).toMatchSnapshot();
    }
  );
});
