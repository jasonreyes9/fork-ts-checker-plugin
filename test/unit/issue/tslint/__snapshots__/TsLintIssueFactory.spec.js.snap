// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] issue/tslint/TsLintIssueFactory creates Issue from TsLint RuleFailure: {"getFailure": [Function getFailure], "getFileName": [Function getFileName], "getRuleName": [Function getRuleName], "getRuleSeverity": [Function getRuleSeverity], "getStartPosition": [Function getStartPosition]} 1`] = `
Object {
  "character": 3,
  "code": "no-unused-vars",
  "file": "src/test.ts",
  "line": 3,
  "message": "'x' is assigned a value but never used.",
  "origin": "tslint",
  "severity": "warning",
}
`;

exports[`[UNIT] issue/tslint/TsLintIssueFactory creates Issue from TsLint RuleFailure: {"getFailure": [Function getFailure], "getFileName": [Function getFileName], "getRuleName": [Function getRuleName], "getRuleSeverity": [Function getRuleSeverity], "getStartPosition": [Function getStartPosition]} 2`] = `
Object {
  "character": 15,
  "code": "no-unused-vars",
  "file": "src/index.ts",
  "line": 11,
  "message": "'y' is assigned a value but never used.",
  "origin": "tslint",
  "severity": "error",
}
`;

exports[`[UNIT] issue/tslint/TsLintIssueFactory creates Issues from TsLint RuleFailures: [[Object], [Object]] 1`] = `
Array [
  Object {
    "character": 15,
    "code": "no-unused-vars",
    "file": "src/index.ts",
    "line": 11,
    "message": "'y' is assigned a value but never used.",
    "origin": "tslint",
    "severity": "error",
  },
  Object {
    "character": 3,
    "code": "no-unused-vars",
    "file": "src/test.ts",
    "line": 3,
    "message": "'x' is assigned a value but never used.",
    "origin": "tslint",
    "severity": "warning",
  },
]
`;
